internal/provider/serverless_exports_data_source.go

```diff
@@ -0,0 +1,384 @@
+									Attributes: map[string]schema.Attribute{
+										"uri": schema.StringAttribute{
+											MarkdownDescription: "The GCS URI of the export target.",
+											Computed:            true,
+										},
+										"auth_type": schema.StringAttribute{
+											MarkdownDescription: "The auth method of the export target.",
+											Computed:            true,
+										},
+									},
+								},
+								"azure_blob": schema.SingleNestedAttribute{
+									MarkdownDescription: "Azure Blob target.",
+									Computed:            true,
+									Attributes: map[string]schema.Attribute{
+										"uri": schema.StringAttribute{
+											MarkdownDescription: "The Azure Blob URI of the export target.",
+											Computed:            true,
+										},
+										"auth_type": schema.StringAttribute{
+											MarkdownDescription: "The auth method of the export target.",
+											Computed:            true,
+										},
+									},
+								},
+							},
+						},
+						"reason": schema.StringAttribute{
+							MarkdownDescription: "The failed reason of the export.",
+							Computed:            true,
+						},
+					},
+				},
+			},
+		},
+	}
+}
+
+func (d *serverlessExportsDataSource) Read(ctx context.Context, req datasource.ReadRequest, resp *datasource.ReadResponse) {
+	var data serverlessExportsDataSourceData
+	diags := req.Config.Get(ctx, &data)
+	resp.Diagnostics.Append(diags...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	tflog.Trace(ctx, "read serverless exports data source")
+	exports, err := d.retrieveExports(ctx, data.ClusterId.ValueString())
+	if err != nil {
+		resp.Diagnostics.AddError("Read Error", fmt.Sprintf("Unable to call ListExports, got error: %s", err))
+		return
+	}
+	var items []serverlessExportItem
+	for _, export := range exports {
+		var e serverlessExportItem
+		e.DisplayName = types.StringValue(*export.DisplayName)
+		e.State = types.StringValue(string(*export.State))
+		e.CreateTime = types.StringValue(export.CreateTime.String())
+		e.CreatedBy = types.StringValue(*export.CreatedBy)
+		if export.Reason.IsSet() {
+			e.Reason = types.StringValue(*export.Reason.Get())
+		}
+		if export.UpdateTime.IsSet() {
+			e.UpdateTime = types.StringValue(export.UpdateTime.Get().String())
+		}
+		if export.CompleteTime.IsSet() {
+			e.CompleteTime = types.StringValue(export.CompleteTime.Get().String())
+		}
+		if export.SnapshotTime.IsSet() {
+			e.SnapshotTime = types.StringValue(export.SnapshotTime.Get().String())
+		}
+		if export.ExpireTime.IsSet() {
+			e.ExpireTime = types.StringValue(export.ExpireTime.Get().String())
+		}
+
+		exportOptionsFileType := *export.ExportOptions.FileType
+		eo := exportOptions{
+			FileType: types.StringValue(string(exportOptionsFileType)),
+		}
+		if export.ExportOptions.Filter != nil {
+			if export.ExportOptions.Filter.Sql != nil {
+				eo.Filter = &exportFilter{
+					Sql: types.StringValue(*export.ExportOptions.Filter.Sql),
```

Comment:
this part is same as refreshServerlessExportDataSourceData right? could we re-use it? maybe try generics

Reply:
okay